platform/linuxbsd/wayland/wayland_thread.h

```diff
@@ -427,6 +453,10 @@ class WaylandThread {
 		struct wl_seat *wl_seat = nullptr;
 		uint32_t wl_seat_name = 0;
 
+		// Touch.
+		struct wl_touch *wl_touch = nullptr;
+		TouchData touch_data_buffer;
```

Comment:
now, regarding that "frame" stuff: the pointer stuff has two `PointerData`, one with the committed data and one without. While you could do the same here, I think that this might make data comparison a lot more annoying.

Because of this, it might make more sense to move "buffering" to each touch point and iterate over them, I think.